e2e_tests/tests/nightly/compute_stats.py

```diff
@@ -15,7 +15,20 @@
 
 
 def iso_date_to_epoch(iso_date: str) -> int:
-    return calendar.timegm(datetime.strptime(iso_date[:-4], DATE_PATTERN).timetuple())
+    iso_format = re.compile(r"^\d{4}-\d{2}-\d{2}\S\d{2}:\d{2}:\d{2}\.\d{3,}[-|\+]\d{2}:\d{2}$")
```

Comment:
Perhaps use `dateutil.parser` instead of doing custom parsing, e.g.

```
from dateutil import parser
...
return int(parser.parse(iso_date).timestamp())
```